ci: automate release merge

rjsparks opened the pull request.

rjsparks committed:
feat(ci): automate release merge

rjsparks committed:
fix: do not delete branches

jennifer-richards commented:
Is the "do not delete branches" step expected to be permanent, or is that a while-debugging/testing feature that will change back?

NGPixel committed:
chore: Upgrade create-pull-request action

Updated the create-pull-request action version from v6 to v7 and fixed a typo in the commit title for the main to release PR.
